hand-written.md

```diff
@@ -1,145 +1,89 @@
 
-As of M3, this feature is not yet on by default.  You can enable it with the `-Xexperimental` compiler option.
+Scala 2.12 emits closures in the same style as Java 8, whether they target a FunctionN class from the standard library or a user-defined Single Abstract Method type. The type checker accepts a function literal as a valid expression for either kind of "function-like" type (built-in or SAM). This improves the experience of using libraries written for Java 8 in Scala.
 
-When the option is enabled, then similar to Java 8, Scala 2.12 allows instantiating any type with one single abstract method by passing a lambda.  This improves the experience of using libraries written for Java 8 in Scala.
+For each lambda the compiler generates a method containing the lambda body, and emits an `invokedynamic` that will spin up a lightweight class for this closure using the JDK's `LambdaMetaFactory`.
 
-This feature was also available in Scala 2.11, also via `-Xexperimental`.
+Compared to Scala 2.11, the new scheme has the advantage that, in most cases, the compiler does not need to generate an anonymous class for each closure. This leads to significantly smaller JAR files.
 
-#### New tab-completion in the Scala REPL
+#### New back end
 
-The implementation of tab-completion in the Scala REPL has been rewritten and now uses the same infrastructure as for example the Scala IDE or Ensime. Note that this feature will also be available in 2.11.8.
+Scala 2.12 standardizes on the "GenBCode" back end, which emits code more quickly because it directly generates ASM bytecode from Scala compiler trees, while the previous back end used an intermediate representation called "ICode". The old back ends (GenASM and GenIcode) have been removed ([#4814](https://github.com/scala/scala/pull/4814), [#4838](https://github.com/scala/scala/pull/4838)).
 
-There are a number of improvements:
-* Reliable completion, also in partial expressions and syntactically incorrect programs: try `class C { def f(l: List[Int]) = l.<TAB>`
-* CamelCase completion: try `(l: List[Int]).rro<TAB>`, it expands to `(l: List[Int]).reduceRightOption`
-* Show desugarings performed by the compiler by adding `//print`: try `for (x <- 1 to 10) println(x) //print<TAB>`
-* Complete bean getters without typing `get`: try `(d: java.util.Date).day<TAB>`
-* Find members by typing any CamelCased part of the name: try `classOf[String].typ<TAB>` to get `getAnnotationsByType`, `getComponentType` and others
-* Complete non-qualified names, including types: try `def f(s: Str<TAB>`
-* Press tab twice to see the method signature: try `List(1,2,3).part<TAB>`, which completes to `List(1,2,3).partition`; press tab again to display `def partition(p: Int => Boolean): (List[Int], List[Int])`
-
-Thanks to @som-snytt for helping out with this work!
 
 #### New bytecode optimizer
 
-The GenBCode backend includes a new inliner and bytecode optimizer.
+The GenBCode back end includes a new inliner and bytecode optimizer.
```

Comment:
in this PR, but i see now that all of those are actually in lines being deleted. so either way is fine with me, i'm just more used to a single word. another example is http://llvm.org/docs/WritingAnLLVMBackend.html